File: ruby/udemy_class/stacks_deques_and_queues/stack.rb
class Stack
  def initialize(store = [])
    # Allow us to start a stack that is not empty
    @store = store
  end

  def push(value)
    @store << value
  end

  def pop
    @store.pop
  end

  def peek
    @store[-1]
  end

  def size
    @store.length
  end

  def is_empty?
    @store.empty?
  end
end

stack = Stack.new
puts stack.is_empty?
stack.push(1)
puts stack.peek
stack.push(2)
puts stack.peek
stack.pop
puts stack.peek
puts stack.is_empty?

File: python/udemy_class/stacks_deques_and_queues/stack.py
class Stack:

    def __init__(self, store=[]):
        self.store = []

    def push(self, value):
        return self.store.append(value)

    def pop(self):
        return self.store.pop()

    def peek(self):
        return self.store[-1]

    def is_empty(self):
        return self.size() == 0

    def size(self):
        return len(self.store)

stack = Stack()

print(stack.is_empty())
stack.push(1)
print(stack.peek())
stack.push(2)
print(stack.peek())
stack.pop()
print(stack.peek())
print(stack.is_empty())
